operator/internal/controller/clustertopology/reconciledelete_test.go

```diff
@@ -0,0 +1,338 @@
+			clusterTopologyName: "test-topology",
+			topologyConfig: configv1alpha1.TopologyConfiguration{
+				Enabled: false,
+			},
+			existingPodCliqueSets: []client.Object{
+				testutils.NewPodCliqueSetBuilder("test-pcs", "test-ns", uuid.NewUUID()).
+					WithTopologyLabel("test-topology").
+					Build(),
+			},
+			wantRequeue: true,
+			wantError:   false,
+		},
+		{
+			name:                "blocked by multiple PodCliqueSet references",
+			clusterTopologyName: "test-topology",
+			topologyConfig: configv1alpha1.TopologyConfiguration{
+				Enabled: false,
+			},
+			existingPodCliqueSets: []client.Object{
+				testutils.NewPodCliqueSetBuilder("test-pcs-1", "test-ns", uuid.NewUUID()).
+					WithTopologyLabel("test-topology").
+					Build(),
+				testutils.NewPodCliqueSetBuilder("test-pcs-2", "test-ns", uuid.NewUUID()).
+					WithTopologyLabel("test-topology").
+					Build(),
+				testutils.NewPodCliqueSetBuilder("test-pcs-3", "other-ns", uuid.NewUUID()).
+					WithTopologyLabel("test-topology").
+					Build(),
+			},
+			wantRequeue: true,
+			wantError:   false,
+		},
+		{
+			name:                "blocked by topology enabled with default name",
+			clusterTopologyName: "grove-topology",
+			topologyConfig: configv1alpha1.TopologyConfiguration{
+				Enabled: true,
+				Name:    nil, // defaults to grove-topology
+			},
+			existingPodCliqueSets: []client.Object{},
+			wantRequeue:           true,
+			wantError:             false,
+		},
+		{
+			name:                "blocked by topology enabled with custom name",
+			clusterTopologyName: "custom-topology",
+			topologyConfig: configv1alpha1.TopologyConfiguration{
+				Enabled: true,
+				Name:    ptr.To("custom-topology"),
+			},
+			existingPodCliqueSets: []client.Object{},
+			wantRequeue:           true,
+			wantError:             false,
+		},
+		{
+			name:                "allowed when topology enabled but different name",
+			clusterTopologyName: "grove-topology",
+			topologyConfig: configv1alpha1.TopologyConfiguration{
+				Enabled: true,
+				Name:    ptr.To("other-topology"),
+			},
+			existingPodCliqueSets: []client.Object{},
+			wantRequeue:           false,
+			wantError:             false,
+		},
+		{
+			name:                "topology disabled - no references",
+			clusterTopologyName: "test-topology",
+			topologyConfig: configv1alpha1.TopologyConfiguration{
+				Enabled: false,
+			},
+			existingPodCliqueSets: []client.Object{},
+			wantRequeue:           false,
+			wantError:             false,
+		},
+		{
+			name:                "PodCliqueSet with different topology label - allowed",
+			clusterTopologyName: "test-topology",
+			topologyConfig: configv1alpha1.TopologyConfiguration{
+				Enabled: false,
+			},
+			existingPodCliqueSets: []client.Object{
+				testutils.NewPodCliqueSetBuilder("test-pcs", "test-ns", uuid.NewUUID()).
+					WithTopologyLabel("different-topology").
+					Build(),
+			},
+			wantRequeue: false,
+			wantError:   false,
+		},
+	}
```

Comment:
Consider adding a test case where the topology is enabled and the name is explicitly set to `ptr.To("grove-topology")` (not nil) to ensure both code paths (nil default vs explicit value) work identically.